ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -370,157 +358,141 @@ S3Open(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session, const char *name,
- *     contains the AWS access key ID and the AWS secret key as comma-separated values.
- */
+// Return a customized file system to access the s3 storage source objects. The authToken
+// contains the AWS access key ID and the AWS secret key as comma-separated values.
 static int
 S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, const char *bucket,
   const char *authToken, const char *config, WT_FILE_SYSTEM **fileSystem)
 {
-    S3_FILE_SYSTEM *fs;
-    WT_FILE_SYSTEM *wtFileSystem;
-    S3_STORAGE *s3;
+    S3Storage *s3;
     int ret;
-    std::string cacheDir;
 
-    s3 = (S3_STORAGE *)storageSource;
+    s3 = (S3Storage *)storageSource;
+
+    // We need to have a bucket to setup the file system. The bucket is expected to be a name and a
+    // region, separated by a semi-colon. eg: 'abcd;ap-southeast-2'.
 
-    /*
-     * We need to have a bucket to setup the file system. The bucket is expected to be a name and a
-     * region, separated by a semi-colon. eg: 'abcd;ap-southeast-2'.
-     */
     if (bucket == NULL || strlen(bucket) == 0) {
         s3->log->LogErrorMessage("S3CustomizeFileSystem: bucket not specified.");
         return (EINVAL);
     }
     int delimiter = std::string(bucket).find(';');
     if (delimiter == std::string::npos || delimiter == 0 || delimiter == strlen(bucket) - 1) {
         s3->log->LogErrorMessage(
-          "S3CustomizeFileSystem: bucket malformed, "
-          "should be a name and a region separated by a semi-colon.");
+          "S3CustomizeFileSystem: improper bucket name, "
+          "should be a name and a region separated by a semicolon.");
         return (EINVAL);
     }
     const std::string bucketName = std::string(bucket).substr(0, delimiter);
     const std::string region = std::string(bucket).substr(delimiter + 1);
 
-    /* Fail if there is no authentication provided. */
+    // Fail if there is no authentication provided.
     if (authToken == NULL || strlen(authToken) == 0) {
         s3->log->LogErrorMessage("S3CustomizeFileSystem: authToken not specified.");
         return (EINVAL);
     }
 
-    /*
-     * An auth token is needed to setup the file system. The token is expected to be an access key
-     * and a secret key separated by a semi-colon.
-     */
+    // An auth token is needed to setup the file system. The token is expected to be an access key
+    // and a secret key separated by a semi-colon.
     if (authToken == NULL || strlen(authToken) == 0) {
         s3->log->LogErrorMessage("S3CustomizeFileSystem: auth token not specified.");
         return (EINVAL);
     }
     delimiter = std::string(authToken).find(';');
     if (delimiter == std::string::npos || delimiter == 0 || delimiter == strlen(authToken) - 1) {
-        s3->log->LogErrorMessage("S3CustomizeFileSystem: authToken malformed.");
+        s3->log->LogErrorMessage("S3CustomizeFileSystem: improper authToken.");
```

Comment:
Maybe we should help the user by telling them the expected form, like we do for the bucket name.

Reply:
Fixed.